node/src/components/consensus/highway_core/test_macros.rs

```diff
@@ -86,6 +87,30 @@ macro_rules! add_vote {
             seq_number,
             timestamp: ($time).into(),
             round_exp: $round_exp,
+            endorsed: vec![],
+        };
+        let hash = wvote.hash();
+        let swvote = SignedWireVote::new(wvote, &TestSecret(($creator).0), &mut $rng);
+        $state.add_vote(swvote).map(|()| hash)
```

Comment:
Can this whole case be reduced to the next case (by just adding a semicolon?), and deduplicated?
```rust
add_vote!($state, $rng, $creator, $time, $round_exp, $val; $($obs),*;)
```

Reply:
I didn't want to replace all the 28 places where the `add_vote!`, without endorsements, is being called. I will see if your suggestion achieves that.

Reply:
No, that's not what I mean!
I just mean replacing the _implementation_ of this case with that single line I wrote.

Reply:
A macro can call itself.

Reply:
OK, it almost worked. I had to pass `vec![]` after the semicolon.